src/Package/Fetch.zig

```diff
@@ -461,7 +461,31 @@ fn runResource(
         };
         defer tmp_directory.handle.close();
 
-        try unpackResource(f, resource, uri_path, tmp_directory);
+        const package_dir = try unpackResource(f, resource, uri_path, tmp_directory);
+
+        if (package_dir) |dir_name| {
+            // Position tmp_directory to dir_name inside tmp_dir_sub_path.
+            const path = try cache_root.join(arena, &.{ tmp_dir_sub_path, dir_name });
+            const handle = tmp_directory.handle.openDir(dir_name, .{ .iterate = true }) catch |err| {
```

Comment:
No need to open the sub directory since the parent is already open. The `Path` abstraction already has a `sub_path` field. It looks like you may need to update `computeHash` to accept `Path` rather than `Directory`.